### ISSUE TITLE
Question: how to "hydrate" hookstate store on client side with SSR

### ISSUE THREAD

**Issue by joompit** [avkonst/hookstate#52]
Hello,

First of all, thank you for this library. I'm looking forward to making the switch from Redux to Hookstate since it seems much to have much less boilerplate code, and I believe it will be easier to separate between the different modules.

However, I'm not totally sure how to accomplish what I currently have. You'll have to excuse myself for maybe not being able to explain perfectly what I want to do, please bear with me.

My current setup is with Next.js and Redux, where I'm using the Redux Provider in order to connect the global store. This way, I'm able to, for example, build the store server-side and then reuse it without having to build the store again client-side. I cannot wrap my mind around how to do it with Hookstate, however.

I understand that Next.js is probably a very-specific case scenario, however, it's what I'm using and I would be interested on some pointers on how I could accomplish doing such thing when using SSR. If I were to know what the recommended way would be, I will do my best to implement it for Next.js.

Sorry if my message was confusing. If necessary, I can add the specific configuration I currently have for my setup.

Thank you

**Comment by avkonst** [comment]
When you 'say build store' what do you mean?  Why is it a problem to build on client side? What redux does special about it?

**Comment by joompit** [comment]
I'm sorry if I didn't explain myself correctly. When I wrote "build store" I mean for example the following situation:

A) Server side we call an API (let's say an authentication API) and fill up the authentication store and use it for certain application logic.
B) Server sends the serialized redux data over the network along with the HTML markup.
C) Client side we wouldn't have to call the API to fill up the client side store once again since it was already done server side. It simply deserializes the redux data.

It's certainly not a problem to rebuild the store client side, but it would be more performant to be able to reuse server side in order to avoid calling multiple times the same API.

Maybe the way to go is doing something such as Next.js does with Redux which is serializing and deserializing the object structure (in this case, the hookstate)?

**Comment by avkonst** [comment]
OK. I have read this article: https://redux.js.org/recipes/server-rendering
I never used it myself, but I understand what they are doing. Would you like to do the same thing with Hookstate?

**Comment by joompit** [comment]
Exactly, that would be what I'm aiming for. There is a "plugin" for Next.js which does something like this for Redux: https://github.com/kirill-konshin/next-redux-wrapper (specific code for the plugin is on https://github.com/kirill-konshin/next-redux-wrapper/blob/master/packages/wrapper/src/index.tsx). 

If I understand correctly, I believe that what I need to do is something along the lines of what you do with the **Persistence** plugin (https://github.com/avkonst/hookstate/blob/master/plugins/Persistence/src/Persistence.ts). On that plugin, you do both serialization/deserialization and rebuild the state from another source (in this case, localStorage).

I would only need to make sure to send these states serialized, and then on client-side make sure to set the state to that specific value. 

I believe Hookstate is more versatile and way simpler than Redux, so I'll try to find a way to mimic this same functionality that is available for Redux to Hookstate in order to be able to continue using SSR for my apps and start using Hookstate as well! It might take me a while since I'm not that experienced, but I will try to do it.
